lighthouse-core/report/report-generator.js

```diff
@@ -149,13 +149,44 @@ class ReportGenerator {
   }
 
   /**
+   * Returns aggregations directly from results or maps Config v2 categories to equivalent Config
+   * v1 aggregations.
+   * @param {{aggregations: !Array<!Object>, reportCategories: !Array<!Object>}} results
+   * @param {boolean=} forceV2
+   * @return {!Array<!Aggregation>}
+   */
+  _getAggregations(results, forceV2 = false) {
+    if (results.aggregations.length && !forceV2) {
+      return results.aggregations;
+    }
+
+    return results.reportCategories.map(category => {
+      const name = category.name;
+      const description = category.description;
+
+      return {
+        name, description,
+        categorizable: false,
+        scored: category.id === 'pwa',
+        total: category.score / 100,
+        score: [{
+          name, description,
+          overall: category.score / 100,
+          subItems: category.audits.map(child => child.result),
+        }],
+      };
+    });
+  }
+
+  /**
    * Generates the Lighthouse report HTML.
    * @param {!Object} results Lighthouse results.
    * @param {!string} reportContext What app is requesting the report (eg. devtools, extension)
    * @param {?Object} reportsCatalog Basic info about all the reports to include in left nav bar
+   * @param {boolean=} forceV2
    * @return {string} HTML of the report page.
    */
-  generateHTML(results, reportContext = 'extension', reportsCatalog) {
+  generateHTML(results, reportContext = 'extension', reportsCatalog, forceV2) {
```

Comment:
can we make a separate method like `generateDOMReport`? this will duplicate a bunch of this method but will make it easier to drop the old when we are ready to switch.
will kill boolean trap, too. yay :)

(also i kinda prefer using "dom" as the key word here. v2 can get overloaded and not sure which v2 talking about: config or report UI or report generator etc etc)